### PR TITLE
[JUJU-365] Local deps package.

### PR DESCRIPTION
**Pull request by SimonRichardson**
Allow the building of jujud and jujuc with the use of cgo, but start the process
of ensuring that we don't require juju to use cgo.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,19 +9,17 @@ GOARCH=$(shell go env GOARCH)
...
-CGO_CFLAGS=-I$(GOPATH)/deps/raft/include/ -I$(GOPATH)/deps/dqlite/include/ -I$(GOPATH)/deps/sqlite/
-CGO_LDFLAGS=-L$(GOPATH)/deps/raft/.libs -L$(GOPATH)/deps/dqlite/.libs/ -L$(GOPATH)/deps/sqlite/.libs/ -L$(GOPATH)/deps/libuv/.libs -L$(GOPATH)/deps/lz4/lib -luv -lraft -ldqlite -llz4
-LD_LIBRARY_PATH=$(GOPATH)/deps/dqlite/.libs/:$(GOPATH)/deps/sqlite/.libs/
-CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"
-
+DEPS_DIR ?= $(PROJECT_DIR)/_deps
 BUILD_DIR ?= $(PROJECT_DIR)/_build
 BIN_DIR ?= ${BUILD_DIR}/${GOOS}_${GOARCH}/bin
 
+CGO_CFLAGS=-I$(GOPATH)/_deps/raft/include/ -I$(GOPATH)/_deps/dqlite/include/ -I$(GOPATH)/_deps/sqlite/
```
Comment (SimonRichardson) [inline Makefile:16]: These should be the `DEPS_DIR` and have the includes.

Comment (achilleasa) [reply Makefile]: I will push an update for this in a few min.

**Comment by achilleasa** [comment]
We need to add a pre-flight check for `cgo-go-build` to ensure that we have set up the required include file symlinks (excluding `queue.h`) for musl on the host where jujud is being compiled.

Also, I spotted warnings like ```
/usr/bin/ld: warning: libc.so, needed by .../golang/src/github.com/juju/juju/_deps/juju-dqlite-static-lib-deps/libuv.so, not found (try using -rpath or -rpath-link)
```
The warning doesn't affect the build which completes successfully.

**Review by achilleasa — APPROVED** [review]
As discussed, the bootstrap will fail after my commit for filtering out non-cloud-local API addresses when detecting the local dqlite address. However, we can fix that on a separate card (JUJU-400)

**Pull request merged**